File: vms/public/js/item.js
frappe.ui.form.on('Item', {
    custom_length: function(frm) {
        calculate_volume(frm);
    },
    custom_width: function(frm) {
        calculate_volume(frm);
    },
    custom_height: function(frm) {
        calculate_volume(frm);
    }
});

function calculate_volume(frm) {
    const length_cm = flt(frm.doc.custom_length);
    const width_cm = flt(frm.doc.custom_width);
    const height_cm = flt(frm.doc.custom_height);

    if (length_cm && width_cm && height_cm) {
        const volume_m3 = (length_cm * width_cm * height_cm) / 1000000;
        frm.set_value('custom_volume_per_unit', volume_m3);
    } else {
        frm.set_value('custom_volume_per_unit', 0);
    }
}

File: vms/events/item.py
import frappe
from frappe.utils import flt

def validate(self,method):
    length_cm = flt(self.custom_length)
    width_cm = flt(self.custom_width)
    height_cm = flt(self.custom_height)

    if length_cm and width_cm and height_cm:
        self.custom_volume_per_case = (length_cm * width_cm * height_cm) / 1000000  # m³
    else:
        self.custom_volume_per_case = 0
